script/sign_upload_linux_pkgs.py

```diff
@@ -0,0 +1,256 @@
+    # Instantiate new requests session, versus reusing the repo session above.
+    # Headers was likely being reused in that session, and not allowing us
+    # to set the Accept header to the below.
+    headers = {'Accept': 'application/octet-stream'}
+    asset_auth_url = asset_url + '?access_token=' + \
+        args.github_token
+    if args.debug:
+        # disable urllib3 logging for this session to avoid showing
+        # access_token in logs
+        logging.getLogger("urllib3").setLevel(logging.WARNING)
+    logging.info("Downloading GitHub release asset: {}".format(
+        asset_url + '/' + filename))
+    try:
+        r = requests.get(asset_auth_url, headers=headers, stream=True)
+    except requests.exceptions.ConnectionError as e:
+        logging.error("Error: Received requests.exceptions.ConnectionError, Exiting...")
+        exit(1)
+    except Exception as e:
+        logging.error("Error: Received exception {},  Exiting...".format(type(e)))
+        exit(1)
+    if args.debug:
+        logging.getLogger("urllib3").setLevel(logging.DEBUG)
+    with open(filename, 'wb') as f:
+        for chunk in r.iter_content(chunk_size=1024):
+            if chunk:
+                f.write(chunk)
+
+    logging.debug(
+        "Requests Response status_code: {}".format(r.status_code))
+    if r.status_code == 200:
+        logging.info("Download successful: Response Status Code {}".format(r.status_code))
+        file_list.append('./' + filename)
+    else:
+        logging.debug(
+            "Requests Response status_code != 200: {}".format(r.status_code))
+
+def rename_file_if_exists(filename, logging):
+    if os.path.isfile(os.path.join(os.getcwd(), filename)):
+        now = datetime.datetime.now()
+        datestring = "{}{}{}.{}{}{}".format(now.year, now.month, now.day, now.hour,
+                     now.minute, now.second)
+        logging.info("Found file \'{}\' already exists, renaming to: \'{}\'.".format(filename,
+                      filename + '.' + datestring))
+        try:
+            os.rename(filename, filename + '.' + datestring)
+        except Exception as e:
+            logging.error("Error: could not rename file {}: {}".format(filename, e))
+
+
+def parse_args():
+    desc = "Download Linux packages from GitHub, sign them, then upload to apt/rpm repositories"
+
+    parser = argparse.ArgumentParser(
+        description=desc, formatter_class=RawTextHelpFormatter)
+    parser.add_argument('-c', '--channel', help='The Brave channel, i.e. \'release\', \'beta\', \'dev\'', required=True)
+    parser.add_argument('-d', '--debug', action='store_true', help='Print debug output')
+    parser.add_argument('-f', '--fingerprint', help='GPG Fingerprint to use for signing packages', required=True)
+    parser.add_argument('-g', '--github_token', help='GitHub token to use for downloading releases', required=True)
+    parser.add_argument('-k', '--gpg_full_key_signature', help='GPG full key signature to use for signing '
```

Comment:
not sure i understand this parameter. is this a signing key? (if so it should be named as such)

Reply:
oh do you mean full key ID? it should be called that, not signature.

Reply:
It is the signing key id. I renamed it and changed all the below places where a fingerprint was used.